Formula/root.rb

```diff
@@ -143,7 +149,7 @@ def caveats
       }
     EOS
     (testpath/"test_compile.bash").write <<~EOS
-      $(root-config --cxx) $(root-config --cflags) $(root-config --libs) $(root-config --ldflags) test.cpp
+      $(root-config --cxx) test.cpp $(root-config --cflags) $(root-config --libs) $(root-config --ldflags)
       ./a.out
```

Comment:
Option 1, add RPATH:
```suggestion
      $(root-config --cxx) test.cpp $(root-config --cflags) $(root-config --libs) $(root-config --ldflags) -Wl,-rpath,#{lib}/root
      ./a.out
```

Option 2: add LD_LIBRARY_PATH:
```suggestion
      $(root-config --cxx) test.cpp $(root-config --cflags) $(root-config --libs) $(root-config --ldflags)
      LD_LIBRARY_PATH="#{lib}/root" ./a.out
```

May just do latter to avoid having to wrap lines exceeding style limit.